src/bindings/python/src/openvino/runtime/utils/data_helpers/data_dispatcher.py

```diff
@@ -132,6 +132,14 @@ def _(
     return {k: to_c_style(v) if is_shared else v for k, v in inputs.items()}
 
 
+@normalize_arrays.register(OVDict)
+def _(
+    inputs: dict,
+    is_shared: bool = False,
+) -> dict:
+    return {k.any_name: to_c_style(v) if is_shared else v for k, v in inputs.items()}
```

Comment:
I am not sure if `any_name` fulfil the promise of putting it on inputs again. Is it defined by model itself that inputs and outputs have the same name? Maybe relying on index integer value would be a better fit?

What is your take on this one? @akuporos

Reply:
Yep, agree

Reply:
done